Dockerfile

```diff
@@ -0,0 +1,37 @@
+FROM osrf/ros:humble-desktop-full
+
+# Install dependencies (taken from devcontainer dockerfile)
+RUN DEBIAN_FRONTEND=noninteractive apt-get update -q && \
+    apt-get update -q && \
+    apt-get install -yq --no-install-recommends \
+    lcov \
+    curl \
+    wget \
+    python3-pip \
+    python-is-python3 \
+    python3-argcomplete \
+    python3-colcon-common-extensions \
+    python3-colcon-mixin \
+    python3-rosdep \
+    libpython3-dev \
+    python3-vcstool \
+    ros-humble-tl-expected && \
+    rm -rf /var/lib/apt/lists/*
+
+# Create ROS workspace
+WORKDIR /ros_ws/src
+
+# Clone driver code
+RUN git clone https://github.com/bdaiinstitute/spot_ros2.git .
+
+# Change gitmodule links to HTTPS to anonymously clone them
+RUN sed -i 's/git@github.com:/https:\/\/github.com\//' .gitmodules
+RUN git submodule update --init --recursive
+
+# Run install script
+RUN /ros_ws/src/install_spot_ros2.sh
+
+# Build packages with Colcon
+WORKDIR /ros_ws/
+RUN . /opt/ros/humble/setup.sh && \
+    colcon build --symlink-install --packages-ignore proto2ros_tests
```

Comment:
@ryan-roche nit: feel free to ignore all `proto2ros*` packages.

Reply:
Also, consider using a `--merge-install` using a standard Linux FHS location (like `/opt` ) as `--install-base`. That'll help the multi-stage Dockerfile.

Reply:
Would you be able to detail how I would implement `--merge-install` for a multi-stage dockerfile? This driver is my first time using ROS2 and its associated tools.

Reply:
> @ryan-roche nit: feel free to ignore all `proto2ros*` packages.

Implemented ced6d8e

Reply:
> Would you be able to detail how I would implement --merge-install for a multi-stage dockerfile? This driver is my first time using ROS2 and its associated tools.

Use `--merge-install` instead of `--symlink-install`, set `--install-base` to some system level path like `/opt/application`, then `COPY` over on a second stage (https://docs.docker.com/build/building/multi-stage may be informative).